applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py

```diff
@@ -173,6 +173,26 @@ def _create_linear_solver(self):
                 KM.Logger.PrintWarning("Using MixedULMLinearSolver, definition of ALM parameters recommended")
                 name_mixed_solver = self.contact_settings["mixed_ulm_solver_parameters"]["solver_type"].GetString()
                 if (name_mixed_solver == "MixedULMLinearSolver"):
+                    linear_solver_name = self.settings["linear_solver_settings"]["solver_type"].GetString()
+                    if (linear_solver_name == "AMGCL" or linear_solver_name == "AMGCLSolver"):
+                        amgcl_param = KM.Parameters("""
+                        {
+                            "solver_type" : "AMGCL",
+                            "smoother_type":"ilu0",
+                            "krylov_type": "lgmres",
+                            "coarsening_type": "aggregation",
+                            "max_iteration": 100,
+                            "provide_coordinates": false,
+                            "gmres_krylov_space_dimension": 100,
+                            "verbosity" : 1,
+                            "tolerance": 1e-6,
+                            "scaling": false,
+                            "block_size": 3,
```

Comment:
i am approving, however please note that if you are in 2D the block size should be 2

Reply:
True

Reply:
Done